src/Microsoft.AspNet.Mvc.Core/Rendering/Html/DefaultHtmlGenerator.cs

```diff
@@ -746,18 +725,136 @@ public virtual TagBuilder GenerateAntiForgery([NotNull] ViewContext viewContext)
                 modelExplorer.Metadata,
                 _metadataProvider,
                 viewContext.HttpContext.RequestServices);
-            
+
             var validatorProviderContext = new ModelValidatorProviderContext(modelExplorer.Metadata);
             validatorProvider.GetValidators(validatorProviderContext);
 
             var validators = validatorProviderContext.Validators;
 
-            return 
+            return
                 validators
                 .OfType<IClientModelValidator>()
                 .SelectMany(v => v.GetClientValidationRules(validationContext));
         }
 
+        /// <inheritdoc />
+        public virtual IReadOnlyCollection<string> GetCurrentValues(
+            [NotNull] ViewContext viewContext,
+            ModelExplorer modelExplorer,
+            string expression,
+            bool allowMultiple)
+        {
+            var fullName = GetFullHtmlFieldName(viewContext, expression);
+            if (string.IsNullOrEmpty(fullName))
+            {
+                throw new ArgumentException(Resources.ArgumentCannotBeNullOrEmpty, nameof(expression));
+            }
+
+            var type = allowMultiple ? typeof(string[]) : typeof(string);
+            var rawValue = GetModelStateValue(viewContext, fullName, type);
+
+            // If ModelState did not contain a current value, fall back to ViewData- or ModelExplorer-supplied value.
+            if (rawValue == null)
+            {
+                if (modelExplorer == null)
+                {
+                    // Html.DropDownList() and Html.ListBox() helper case.
+                    rawValue = viewContext.ViewData.Eval(expression);
+                    if (rawValue is IEnumerable<SelectListItem>)
+                    {
+                        // This ViewData item contains the fallback selectList collection for GenerateSelect().
+                        // Do not try to use this collection.
+                        rawValue = null;
```

Comment:
Kind of misleading, this bit essentially means `return null` because we immediately short-circuit.